internal/structs/structs.go

```diff
@@ -78,17 +79,21 @@ func GetTagInfo(key reflect.Type) (StructInfo, error) {
 	return getCachedTagInfo(tagInfoCache, key)
 }
 
-func getCachedTagInfo(tagInfoCache map[reflect.Type]StructInfo, key reflect.Type) (StructInfo, error) {
-	if info, found := tagInfoCache[key]; found {
-		return info, nil
+func getCachedTagInfo(tagInfoCache *sync.Map, key reflect.Type) (StructInfo, error) {
+	if data, found := tagInfoCache.Load(key); found {
+		if info, ok := data.(StructInfo); !ok {
+			return StructInfo{}, fmt.Errorf("invalid cache entry, expected type StructInfo, found %T", data)
+		} else {
+			return info, nil
+		}
 	}
 
 	info, err := getTagNames(key)
 	if err != nil {
 		return StructInfo{}, err
 	}
 
-	tagInfoCache[key] = info
+	tagInfoCache.Store(key, info)
```

Comment:
Hello. It's my first time looking at this project. I have a little consideration about sync.Map here.

Between Load (line 83) and Store (line 96), another goroutine could run the same code. In a lot of usecases, it's ok. But, if you think that it can be a problem here, take a look at this:

https://pkg.go.dev/sync#Map.LoadOrStore

Reply:
That is true, and in that case the last one would save the value. That means that both occurred in a cache-miss, but since the return of `getTagNames` should always be the same for the same key, that is not a problem.

I don't see how we could use `LoadOrStore` over here, because in order to do that we should have the calculated value, meaning we should always call `getTagNames`, which would defeat the caching necessity at all.

Makes sense or am I missing something?

Reply:
Yeah, I think using LoadOrStore won't be possible 🤔  and the possibility of multiple cache misses is real but this would only cause a small initial performance issue, I think it is safe to keep as it is now.